bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,762 @@
+                        newIndoorTemp);
+            }
+        }
+
+        DecimalType newOutdoorTemp = command.getOutdoorTemperature();
+        if (newOutdoorTemp == null) {
+            updateStateIfLinked(
+                    new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_OUTDOOR_TEMPERATURE),
+                    UnDefType.UNDEF);
+        } else if (!newOutdoorTemp.equals(this.madokaSettings.getOutdoorTemperature())) {
+            this.madokaSettings.setOutdoorTemperature(newOutdoorTemp);
+            updateStateIfLinked(
+                    new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_OUTDOOR_TEMPERATURE),
+                    newOutdoorTemp);
+        }
+
+    }
+
+    @Override
+    public void receivedResponse(SetPowerstateCommand command) {
+
+        updateStateIfLinked(new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_ONOFF_STATUS),
+                command.getPowerState());
+
+        madokaSettings.setOnOffState(command.getPowerState());
+
+        if (command.getPowerState() == OnOffType.ON) {
+            // Depending on the state
+            if (madokaSettings.getOperationMode() == null) {
+                return;
+            }
+
+            switch (madokaSettings.getOperationMode()) {
+                case AUTO:
+                    updateStateIfLinked(
+                            new ChannelUID(getThing().getUID(),
+                                    DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                            new StringType("Auto"));
+                    break;
+                case HEAT:
+                    updateStateIfLinked(
+                            new ChannelUID(getThing().getUID(),
+                                    DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                            new StringType("Heating"));
+                    break;
+                case COOL:
+                    updateStateIfLinked(
+                            new ChannelUID(getThing().getUID(),
+                                    DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                            new StringType("Cooling"));
+                    break;
+                default: // Other Modes are not [yet] supported
+                    break;
+            }
+        } else {
+            updateStateIfLinked(
+                    new ChannelUID(getThing().getUID(),
+                            DaikinMadokaBindingConstants.CHANNEL_ID_HOMEKIT_CURRENT_HEATING_COOLING_MODE),
+                    new StringType("Off"));
+
+        }
+    }
+
+    @Override
+    public void receivedResponse(SetSetpointCommand command) {
+
+        // The update depends on the mode - so if not set - skip
+        if (this.madokaSettings.getOperationMode() == null) {
```

Comment:
`this.madokaSettings.getOperationMode()` should be cached to a local variable so it can be reused.

Also make similar changes elsewhere.